kratos/includes/properties.h

```diff
@@ -91,16 +88,33 @@ class Properties : public IndexedObject
 
     typedef std::unordered_map<std::size_t, TableType> TablesContainerType; // This is a provisional implmentation and should be changed to hash. Pooyan.
 
+    /// Properties container. A vector set of properties with their Id's as key.
+    typedef PointerVectorSet<Properties, IndexedObject> PropertiesContainerType;
```

Comment:
This should be PointerVector not a set. And I would name it `SubPropertiesContainerType`

Reply:
How I am supposed to search on the vector without the Set, do I need to check manually the Id?

Reply:
You are right. It makes the implementation easier